mutt/file.c

```diff
@@ -930,11 +930,16 @@ time_t mutt_file_decrease_mtime(const char *f, struct stat *st)
  *
  * @warning mutt_file_dirname() returns a static string which must not be free()'d.
  */
-const char *mutt_file_dirname(const char *p)
+char *mutt_file_dirname(const char *p)
 {
-  static char buf[_POSIX_PATH_MAX];
-  mutt_str_strfcpy(buf, p, sizeof(buf));
-  return dirname(buf);
+  char *buf = malloc(sizeof(_POSIX_PATH_MAX));
```

Comment:
Hmm... that doesn't allocate the amount you think it does!

Also, please use `mutt_mem_malloc()`.
It cannot fail, so you don't need to test the return value.